contrib/docker/ckan-entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+set -eu
+
+# URL for the primary database, in the format expected by sqlalchemy (required
+# unless linked to a container called 'db')
+: ${DATABASE_URL:=}
+# URL for solr (required unless linked to a container called 'solr')
+: ${SOLR_URL:=}
+# URL for redis (required unless linked to a container called 'redis')
+: ${REDIS_URL:=}
+# Email to which errors should be sent (optional, default: none)
+: ${ERROR_EMAIL:=}
+
+CONFIG="${CKAN_CONFIG}/ckan.ini"
+
+abort () {
+  echo "$@" >&2
+  exit 1
+}
+
+write_config () {
+  ckan-paster make-config ckan "$CONFIG"
+
+  ckan-paster --plugin=ckan config-tool "$CONFIG" -e \
+      "sqlalchemy.url = ${DATABASE_URL}" \
+      "solr_url = ${SOLR_URL}" \
+      "ckan.redis.url = ${REDIS_URL}" \
+      "ckan.storage_path = /var/lib/ckan" \
+      "email_to = disabled@example.com" \
+      "error_email_from = ckan@$(hostname -f)" \
+      "ckan.site_url = http://localhost:5000"
```

Comment:
Please just add the ones you need, making arbitrary config options override-able from environment vars makes me nervous.